### PR TITLE
filter with non identifier property names

### LINKED ISSUE #1545: [🐛 BUG] Filter on scenario properties in Scenario Selector not working
### What went wrong? 🤔

Using custom filters on scenario properties will filter all scenarios even if the scenario respects the filters.

This issue could be irrelevant depending on what we want tot do in: https://github.com/Avaiga/taipy/issues/1511

### Expected Behavior

We should filter on properties and scenarios that respect the filters should appear in the scenario selector.

### Steps to Reproduce Issue

This comes from an actual use. There are three custom filters added to the scenario selector that corresponds to three properties added to the scenario.

These are the properties:
```python
    scenario.properties['Fix warehouses'] = True
    scenario.properties['Nb of warehouses'] = 5
    scenario.properties['Fixed warehouses'] = ['Louisville']
```

Run the code below, try to filter on the custom filters.

```python
# Import necessary libraries
import pandas as pd
import taipy as tp
from taipy import Config, Scope, Frequency

import datetime as dt


# Function to run a Dataiku scenario
def run_something(input_1, input_2):
    datetime = dt.datetime.now()
    date = dt.date(2018, 1, 1)
    int_var = 10
    string_var = "String"
    return datetime, date, int_var, string_var


data = {"toto": [i for i in range(10_000)], 
        "titi": [2*i for i in range(10_000)],
        "tata": [4*i for i in range(10_000)]}



input_1_cfg = Config.configure_data_node(
    id="input_1_data_node",
    default_data=data,
)


input_2_cfg = Config.configure_data_node(
    id="input_2_data_node",
    default_data=data,
)

datetime_cfg = Config.configure_data_node(id="datetime_data_node")
date_cfg = Config.configure_data_node(id="date_data_node")
int_cfg = Config.configure_data_node(id="int_data_node")
string_cfg = Config.configure_data_node(id="string_data_node")


# Scenario and task configuration in Taipy
scenario_task_cfg = Config.configure_task(
    id="scenario_task",
    function=run_something,
    input=[input_1_cfg, input_2_cfg],
    output=[datetime_cfg, date_cfg, int_cfg, string_cfg]
)

scenario_cfg = Config.configure_scenario(
    id="scenario",
    task_configs=[scenario_task_cfg],
    frequency=Frequency.DAILY)


filter_for_scenario = ["Config id",
                        "Label",
                        "Creation date", 
                        "Cycle label",
                        "Cycle start",
                        "Cycle end",
                        "Primary", 
                        "Tags", 
                        "Number of Warehouses",
                        "Fix warehouses", 
                        "Fixed warehouses"]

# GUI Markdown content
scenario_md = """
<|{scenario}|scenario_selector|filter={filter_for_scenario}|>

<|{scenario}|scenario|>
"""


def add_properties(scenario):
    scenario.properties['Fix warehouses'] = True
    scenario.properties['Number of Warehouses'] = 5
    scenario.properties['Fixed warehouses'] = ['Louisville']
    return scenario

# Main execution block with GUI setup
if __name__ == "__main__":
    tp.Core().run()
    scenario = tp.create_scenario(scenario_cfg)
    scenario = add_properties(scenario)

    tp.Gui(scenario_md).run(title="Bug replication", port=3248)
```
### Browsers

Chrome

### OS

Windows

### Version of Taipy

develop - 7/17/24

### Acceptance Criteria

- [ ] Ensure new code is unit tested, and check code coverage is at least 90%.
- [ ] Create related issue in taipy-doc for documentation and Release Notes.

### Code of Conduct

- [X] I have checked the [existing issues](https://github.com/Avaiga/taipy/issues?q=is%3Aissue+).
- [ ] I am willing to work on this issue (optional)

### PR DESCRIPTION
**Pull request by FredLL-Avaiga**
resolves #1545

deal with values other than string

```
# Import necessary libraries
import datetime as dt

import taipy as tp
from taipy import Config, Frequency


# Function to run a Dataiku scenario
def run_something(input_1, input_2):
    datetime = dt.datetime.now()
    date = dt.date(2018, 1, 1)
    int_var = 10
    string_var = "String"
    return datetime, date, int_var, string_var


data = {"toto": list(range(10_000)),
        "titi": [2*i for i in range(10_000)],
        "tata": [4*i for i in range(10_000)]}



input_1_cfg = Config.configure_data_node(
    id="input_1_data_node",
    default_data=data,
)


input_2_cfg = Config.configure_data_node(
    id="input_2_data_node",
    default_data=data,
)

datetime_cfg = Config.configure_data_node(id="datetime_data_node")
date_cfg = Config.configure_data_node(id="date_data_node")
int_cfg = Config.configure_data_node(id="int_data_node")
string_cfg = Config.configure_data_node(id="string_data_node")


# Scenario and task configuration in Taipy
scenario_task_cfg = Config.configure_task(
    id="scenario_task",
    function=run_something,
    input=[input_1_cfg, input_2_cfg],
    output=[datetime_cfg, date_cfg, int_cfg, string_cfg]
)

scenario_cfg = Config.configure_scenario(
    id="scenario",
    task_configs=[scenario_task_cfg],
    frequency=Frequency.DAILY)


filter_for_scenario = [
    "Config id",
    "Label",
    "Creation date",
    "Cycle label",
    "Cycle start",
    "Cycle end",
    "Primary",
    "Tags",
    "Number_of_Warehouses",
    "Fix_warehouses",
    "Fixed warehouses",
    "Fixed_warehouses",
    "date_wh"
]

# GUI Markdown content
scenario_md = """
<|{scenario}|scenario_selector|filter={filter_for_scenario}|>

<|{scenario}|scenario|>
"""


def add_properties(scenario):
    scenario.properties["Fix_warehouses"] = True
    scenario.properties["Number_of_Warehouses"] = 5
    scenario.properties["Fixed_warehouses"] = ["Louisville"]
    scenario.properties["Fixed warehouses"] = ["Louisville"]
    scenario.properties["date_wh"] = dt.datetime.now()
    return scenario

# Main execution block with GUI setup
if __name__ == "__main__":
    tp.Core().run()
    scenario = tp.create_scenario(scenario_cfg)
    scenario = add_properties(scenario)

    tp.Gui(scenario_md).run(title="1545 Sort by Scenario properties")

```

### CODE REVIEW

File: taipy/gui_core/_adapters.py [hunk 1]
```diff
@@ -244,14 +244,50 @@ def get_hash():
...
+    return next(filter(lambda v: _filter_value(v, operator, val), list_val), None) is not None
+
 
 def _invoke_action(
     ent: t.Any, col: str, col_type: str, is_dn: bool, action: str, val: t.Any, col_fn: t.Optional[str]
 ) -> bool:
     if ent is None:
         return False
+    if not (col_fn or col).isidentifier():
+        _warn(f'Error filtering with "{col_fn or col}": not a valid python identifier.')
```
Comment (FabienLelaquais) [inline taipy/gui_core/_adapters.py]:
```suggestion
        _warn(f'Error filtering with "{col_fn or col}": not a valid Python identifier.')
```

File: taipy/gui_core/_adapters.py [hunk 2]
```diff
@@ -244,14 +244,50 @@ def get_hash():
...
+            try:
+                return type(base_val)(val)
+            except Exception:
+                # forget it
+                pass
+    return val
+
 def _filter_iterable(list_val: Iterable, operator: t.Callable, val: t.Any):
-    return next(filter(lambda v: operator(v, val), list_val), None) is not None
+    if operator is contains:
```
Comment (FabienLelaquais) [inline taipy/gui_core/_adapters.py:271]: 😮

**Review by FabienLelaquais — DISMISSED** [review]
🫤

**Review by FabienLelaquais — DISMISSED** [review]
➕➕➕

**Review by dinhlongviolin1 — DISMISSED** [review]
Lgtm

**Review by namnguyen20999 — DISMISSED** [review]
LGTM

**Review by FabienLelaquais — DISMISSED** [review]
👌

**Review by FabienLelaquais — APPROVED** [review]
👍

**Pull request merged**